core/Decode.cpp

```diff
@@ -125,19 +131,38 @@ namespace olympia
         }
     }
 
-    // process vset settings being forward from execution pipe
-    // for set instructions that depend on register
-    void Decode::process_vset_(const InstPtr & inst)
+    void Decode::updateVcsrs_(const InstPtr & inst)
     {
         VCSRs_.setVCSRs(inst->getVL(), inst->getSEW(), inst->getLMUL(), inst->getVTA());
-        // AVL setting to VLMAX if rs1 = 0 and rd != 0
-        if (inst->getSourceOpInfoList()[0].field_value == 0
-            && inst->getDestOpInfoList()[0].field_value != 0)
+
+        const uint64_t uid = inst->getOpCodeInfo()->getInstructionUniqueID();
+        if ((uid == mavis_vsetvli_uid_) && inst->hasZeroRegSource())
```

Comment:
Isn't the `hasZeroRegSource` already checked before calling this function?

Reply:
Only for `vsetivli` and certain `vsetvli` instructions, but it is also used when a `vsetvl` is completed. See line 164.